receiver/receivercreator/config.go

```diff
@@ -52,7 +52,10 @@ type receiverTemplate struct {
 	// Rule is the discovery rule that when matched will create a receiver instance
 	// based on receiverTemplate.
 	Rule string `mapstructure:"rule"`
-	rule rule
+	// ResourceAttributes is a map of resource attributes to add to just this receiver's resource metrics.
+	// It can contain expr expressions for endpoint env value expansion
+	ResourceAttributes map[string]string `mapstructure:"resource_attributes"`
```

Comment:
As it also happens on #11868, I would suggest changing this to `map[string]interface{}` and erroring out if there are non-string attributes, to ensure that we can support non-string attributes in the future.

Reply:
Updated the config field to be `map[string]interface{}` with additional unit test.